src/Modules/OrchardCore.Commerce/Services/ProductService.cs

```diff
@@ -37,11 +38,42 @@ public async Task<IEnumerable<ProductPart>> GetProductsAsync(IEnumerable<string>
 
         var contentItems = await _contentManager.GetAsync(contentItemIds);
 
-        // We need BuildDisplayAsync to fill part.Elements with the fields. We could extract the logic from
-        // BuildDisplayAsync, but we would have to copy almost everything from there.
+        // We need to replicate some things that BuildDisplayAsync does to fill part.Elements with the fields. We can't
+        // use BuildDisplayAsync directly because it requires a BuildDisplayContext.
         foreach (var contentItem in contentItems)
         {
-            await _contentDisplayHandler.BuildDisplayAsync(contentItem, context: null);
+            var contentItemsPartName = contentItem.ContentType;
```

Comment:
Seems to me that this will only work for the content types created in the Admin UI. Any custom content type is unlikely to have a content type named part. (e.g. Product content type would have a part named ProductPart rather than Product, so this wouldn't find it)

I think you should get every field from all parts, and then filter to only include those that inherit from the `ProductAttributeField` abstract class.